### PR TITLE
fix(groups-control): address rendering issues

### PR DESCRIPTION
**Pull request by stiefeljackal**
This is to address the issue reported in #26 by @bontebok. The issues with the previous group list view control are the following:

* Some group options were going out of bounds; this makes them hidden from being selected unless the user should resize the width of the window. Please see the screenshots in #26.
* Text on some group options were getting cutoff due to how the elements within the data template are organized.

The first issue is addressed by removing the `HorizontalAlignment` property on the `GroupsListView` control as that was the culprit for missing group options. `Margin` was also removed to make the control aligned with the other controls on the Download screen.

The second issue is addressed by removing `StackPanel` that combined the "(Admin)" tag and the name of the group; this prevented the text from wrapping correctly. I have also adjusted the layout parameters to help organize the group options in a better layout than the one before. In addition, a new view model called `GroupsListItemViewModel` was created to represent the text for these checkboxes. The main text will only show up to 60 characters with a trailing ellipses at the end if the 60 character max is reached. Groups will more than likely not have more than 60 characters, but it is there regardless to keep the main title to two lines at max. Regarding the introduction of the new view model, I have reviewed the MVVM design and looked at suggestions from others; most have indicated that a view model with children view models is not an anti-design pattern. In fact, I would even suggest that `ShouldDownload` be relocated to this view model as the model should not contain this property; `ShouldDownload` is a view model property. I would make that change, but since making too many changes seemed to frazzle and confuse the repository maintainer despite that fact that there are semantic commit messages that provide simplistic details, I have decided not to do that and merely just suggest this. :)

Anyway, I hope this helps.

### CODE REVIEW

File: AccountDownloader/ViewModels/Controls/GroupsListItemViewModel.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
+﻿using AccountDownloader.Properties;
+using AccountDownloader.Services;
+using ReactiveUI;
+
+namespace AccountDownloader.ViewModels;
+
+public class GroupsListItemViewModel: ReactiveObject
```
Comment (GuVAnj8Gv3RJ) [inline AccountDownloader/ViewModels/Controls/GroupsListItemViewModel.cs:7]: Implement IGroup plz

File: AccountDownloader/ViewModels/Controls/GroupsListViewModel.cs [hunk 2]
```diff
@@ -24,28 +26,28 @@ public class GroupsListViewModel: ReactiveObject
 
         public GroupsListViewModel(IEnumerable<IGroup>? groups = null)
         {
-            Groups = new ObservableCollection<IGroup>();
```
Comment (GuVAnj8Gv3RJ) [inline AccountDownloader/ViewModels/Controls/GroupsListViewModel.cs:27]: I want to preserve IGroup plz

File: AccountDownloader/ViewModels/Controls/GroupsListItemViewModel.cs [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+{
+    private const byte MAX_TITLE_LENGTH = 60;
+
+    public IGroup Group { get;}
+
+    public string GroupTitleLabel
+    {
+        get
+        {
+            var title  = $"{(Group.IsAdmin ? $"{Resources.GroupsAdminIndicator} " : "")}{Group.Name}";
```
Comment (GuVAnj8Gv3RJ) [inline AccountDownloader/ViewModels/Controls/GroupsListItemViewModel.cs:17]: This all reads a bit cluttered due to nested evaluation. I'll clean it up after its merged.

**Review by GuVAnj8Gv3RJ — CHANGES_REQUESTED** [review]

File: AccountDownloader/ViewModels/Controls/GroupsListItemViewModel.cs (hunk 3 above)
Comment (stiefeljackal) [reply AccountDownloader/ViewModels/Controls/GroupsListItemViewModel.cs:17]: It is a bit cluttered. I can clean that up.

**Comment by stiefeljackal** [comment]
Adding the MIT+GPLv2 License for now until an explanation is given. If we do resolve these issues and come to a compromise, then I will revert that commit back.

**Comment by stiefeljackal** [comment]
And squashing commits with said LICENSE. As said, I will revert the LICENCE back to the original once a compromised is reached if one does.

**Comment by GuVAnj8Gv3RJ** [comment]
We're sorry, we're unable to accept license changes of any kind at this time.

**Comment by stiefeljackal** [comment]
You didn’t even read the comments! I would have taken the license off as long as you gave explanations.

**Comment by stiefeljackal** [comment]
Additionally, I don’t appreciate your tone as far as the comments you made in this code review with the exception of one. Just saying “Plz” without providing a sufficient explanation is unprofessional and rude. If you ask me, I am working with a child here. If you want people to accept you, you must act professional and provide an acceptable argument. What you provided is childish and will distance others from working with you. I want to see you be successful, but you are being suspicious.

**Comment by GuVAnj8Gv3RJ** [comment]
These comments unfortunately further illustrate why I'm unable to accept this contribution.

In industry, it is common to "assume best intentions" which seems to have gone awry with your opinion of me here. As I wish to keep things as professional as possible. We'll unfortunately have to stick to subsequent issues and PRs should you wish to make them.

Thank you.

**Comment by stiefeljackal** [comment]
In industry, I know the person I am working with. In industry, a person would perform a proper code review together and provide an explanation of why something should change. Your poor communication with the first two comments are not what I deemed professional.

Thank you.

P.S. Your communication in the last comment sounded so much better than your code review comments.

**Pull request closed without merging**